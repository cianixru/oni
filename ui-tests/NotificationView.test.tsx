import { mount, shallow, configure } from "enzyme"
import { shallowToJson } from "enzyme-to-json"
import * as React from "react"

import styled from "styled-components"
import {
    Button,
    NotificationContents,
    NotificationDescription,
    NotificationTitle,
    NotificationView,
} from "./../browser/src/Services/Notifications/NotificationsView"

describe("<NotificationsView />", () => {
    const buttons = [
        {
            title: "yes",
            callback: () => ({}),
        },
        {
            title: "no",
            callback: () => ({}),
        },
    ]
    const Notification = (
        <NotificationView
            id="1"
            level="info"
            title="testing1"
            detail="this is a test"
            expirationTime={20000000}
            onClick={() => console.log("clicked")} //tslint:disable-line
            onClose={() => console.log("closed")} //tslint:disable-line
        />
    )

    const NotificationWithButtons = (
        <NotificationView
            id="1"
            level="info"
            title="testing1"
            detail="this is a test"
            expirationTime={20000000}
            onClick={() => console.log("clicked")} //tslint:disable-line
            onClose={() => console.log("closed")} //tslint:disable-line
            buttons={buttons}
        />
    )
    it("It renders a notification component correctly", () => {
        const wrapper = shallow(Notification)
        expect(wrapper.length).toEqual(1)
    })

    it("Renders the correct title", () => {
        const wrapper = shallow(Notification)
        expect(
            wrapper
                .find(NotificationTitle)
                .dive()
                .text(),
        ).toEqual("testing1")
    })

    it("Renders the correct text", () => {
        const wrapper = shallow(Notification)
        expect(
            wrapper
                .find(NotificationContents)
                .dive()
                .find(NotificationDescription)
                .dive()
                .text(),
        ).toEqual("this is a test")
    })

    it("should match the snapshot", () => {
        const wrapper = shallow(Notification)
        expect(shallowToJson(wrapper)).toMatchSnapshot()
    })

    it("Renders buttons with correct text when passed button fields", () => {
        const wrapper = shallow(NotificationWithButtons)
        expect(wrapper.contains([<Button>yes</Button>, <Button>no</Button>]))
    })
})
