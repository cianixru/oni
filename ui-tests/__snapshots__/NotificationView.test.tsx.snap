// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationsView /> should match the snapshot 1`] = `
<styled.div
  className="notification"
  data-test="notification"
  key="1"
  level="info"
  onClick={[Function]}
>
  <styled.header>
    <styled.div>
      <styled.div
        level="info"
      >
        <Sneakable
          callback={[Function]}
        >
          <Icon
            name="info-circle"
            size={1}
          />
        </Sneakable>
      </styled.div>
      <styled.div
        onClick={[Function]}
      >
        <Sneakable
          callback={[Function]}
        >
          <Icon
            name="times"
            size={1}
          />
        </Sneakable>
      </styled.div>
    </styled.div>
    <styled.div
      data-test="notification-title"
      level="info"
    >
      testing1
    </styled.div>
  </styled.header>
  <styled.div>
    <styled.div
      className="notification-description"
    >
      this is a test
    </styled.div>
  </styled.div>
</styled.div>
`;
